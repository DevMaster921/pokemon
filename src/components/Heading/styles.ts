import styled from 'styled-components'
import { HeadingProps } from '.'

export const Wrapper = styled.h1.attrs(
  ({ small, isWhite, toUppercase }: Omit<HeadingProps, 'children'>) => ({
    small,
    isWhite,
    toUppercase
  })
)`
  color: ${({ isWhite }) => (isWhite ? 'white' : 'black')};
  font-size: ${({ small }) => (small ? '1.8em' : '3.5em')};
  margin-bottom: 1em;
  position: relative;
  text-transform: ${({ toUppercase }) => toUppercase && 'uppercase'};

  &::after {
    border-bottom: ${({ isWhite }) =>
      isWhite ? '0.5rem solid white' : '0.5rem solid #d31e25'};
    bottom: -1rem;
    content: '';
    left: 0;
    position: absolute;
    width: 3rem;
  }
`
