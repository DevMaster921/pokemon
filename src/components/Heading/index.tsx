import * as S from './styles'

export type HeadingProps = {
  children: string
  small?: boolean
  isWhite?: boolean
  toUppercase?: boolean
}

const Heading = ({
  children,
  small,
  isWhite = false,
  toUppercase = false
}: HeadingProps) => (
  <S.Wrapper toUppercase={toUppercase} isWhite={isWhite} small={small}>
    {children}
  </S.Wrapper>
)

export default Heading
