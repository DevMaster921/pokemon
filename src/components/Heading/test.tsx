import { screen, render } from '@testing-library/react'

import Heading from '.'

describe('<Heading />', () => {
  it('should render a black heading by default', () => {
    const { container } = render(<Heading>Lorem Ipsum</Heading>)

    expect(screen.getByRole('heading', { name: /Lorem Ipsum/i })).toHaveStyle({
      color: 'black'
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a white heading', () => {
    render(<Heading isWhite>Lorem Ipsum</Heading>)

    expect(screen.getByRole('heading', { name: /Lorem Ipsum/i })).toHaveStyle({
      color: 'white'
    })
  })

  it('should render a small heading when small is passed', () => {
    render(<Heading small>Lorem Ipsum</Heading>)

    expect(screen.getByRole('heading', { name: /Lorem Ipsum/i })).toHaveStyle({
      fontSize: '1.8em'
    })
  })

  it('should render a uppercase heading when toUppercase is passed', () => {
    render(<Heading toUppercase>Lorem Ipsum</Heading>)

    expect(screen.getByRole('heading', { name: /Lorem Ipsum/i })).toHaveStyle({
      textTransform: 'uppercase'
    })
  })
})
