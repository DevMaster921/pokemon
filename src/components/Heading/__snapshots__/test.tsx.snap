// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render a black heading by default 1`] = `
.c0 {
  color: black;
  font-size: 3.5em;
  margin-bottom: 1em;
  position: relative;
}

.c0::after {
  border-bottom: 0.5rem solid #d31e25;
  bottom: -1rem;
  content: '';
  left: 0;
  position: absolute;
  width: 3rem;
}

<h1
  class="c0"
>
  Lorem Ipsum
</h1>
`;
